packages/jest-core/src/SearchSource.ts

```diff
@@ -328,4 +332,29 @@ export default class SearchSource {
 
     return searchResult;
   }
+
+  findRelatedSourcesFromTestsInChangedFiles(
+    changedFilesInfo: ChangedFiles,
+  ): Array<string> {
+    if (!hasSCM(changedFilesInfo)) {
+      return [];
+    }
+    const {changedFiles} = changedFilesInfo;
+    const dependencyResolver = new DependencyResolver(
```

Comment:
The same DependencyResolver is also created for `findRelatedTests`. Can we cache it then?